specification/subscription/resource-manager/Microsoft.Subscription/preview/2021-10-01-preview/subscriptions.json

```diff
@@ -0,0 +1,1156 @@
+      "description": "The parameters required to create a new subscription.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PutAliasRequestProperties",
+          "description": "Put alias request properties."
+        }
+      }
+    },
+    "PutAliasRequestProperties": {
+      "description": "Put subscription properties.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "The friendly name of the subscription.",
+          "type": "string"
+        },
+        "workload": {
+          "$ref": "#/definitions/Workload"
+        },
+        "billingScope": {
+          "$ref": "#/definitions/BillingScope"
+        },
+        "subscriptionId": {
+          "description": "This parameter can be used to create alias for existing subscription Id",
+          "type": "string"
+        },
+        "resellerId": {
+          "description": "Reseller Id",
+          "type": "string"
+        },
+        "additionalProperties": {
+          "$ref": "#/definitions/PutAliasRequestAdditionalProperties",
+          "description": "Put alias request additional properties."
+        }
+      }
+    },
+    "PutAliasRequestAdditionalProperties": {
+      "description": "Put subscription additional properties.",
+      "type": "object",
+      "properties": {
+        "managementGroupId": {
+          "description": "Management group Id for the subscription.",
+          "type": "string"
+        },
+        "subscriptionTenantId": {
+          "description": "Tenant Id of the subscription",
+          "type": "string"
+        },
+        "subscriptionOwnerId": {
+          "description": "Owner Id of the subscription",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Tags for the subscription",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SubscriptionAliasResponse": {
+      "description": "Subscription Information with the alias.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Fully qualified ID for the alias resource.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "name": {
+          "description": "Alias ID.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type, Microsoft.Subscription/aliases.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/SubscriptionAliasResponseProperties",
+          "description": "Subscription Alias response properties."
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
```

Comment:
I hope you could probably inherit ProxyResoruce or Resource from common-types, instead of defining all these fields, do you see any barriers to doing that?